index.js

```diff
@@ -249,38 +254,40 @@ checkElevated()
             }
         }
 
-        return distro()
-            .catch(() => {});
+        return distro().catch(() => {});
     })
-    .then(dist => {
+    .then((dist) => {
         if (dist && dist.os) {
             config.osdist = dist.os + (dist.release ? ' ' + dist.release : '');
         }
-        return new Promise(resolve => {
-            request('http://169.254.169.254/latest/dynamic/instance-identity/document/', { timeout: 1000, json: true }, function(err, res, body) {
+
+        return new Promise((resolve) => {
+            request(config.AWS_INSTANCE_CHECK_URL, {
+                timeout: 1000
+                , json: true
+            }, (err, res, body) => {
                 if (res && res.statusCode && body) {
                     config.awsid = body.instanceId;
                     config.awsregion = body.region;
                     config.awsaz = body.availabilityZone;
                     config.awsami = body.imageId;
                     config.awstype = body.instanceType;
                 }
-                resolve(macaddress.allAsync()
-                    .catch(() => {})
-                );
+                resolve(macaddress.allAsync().catch(() => {}));
```

Comment:
do you want to `reject` when there is an error?

Reply:
`macaddress` is optional nice-to-have here; therefore, no need to `reject` in the case of `error`.